src/datetimesettings.cpp

```diff
@@ -100,40 +139,72 @@ void DateTimeSettings::setTimezone(const QString &tz)
         return;
     }
 
-    m_time.setTimezone(tz);
+    Maemo::Timed::WallClock::Settings s;
+    s.setTimezoneManual(tz);
+    setSettings(s);
 }
 
 void DateTimeSettings::setHourMode(DateTimeSettings::HourMode mode)
 {
-    Maemo::Timed::Interface timedInterface;
-    Maemo::Timed::WallClock::Settings settings;
-    settings.setFlag24(mode == TwentyFourHours);
-    timedInterface.wall_clock_settings_async(settings);
+    Maemo::Timed::WallClock::Settings s;
+    s.setFlag24(mode == TwentyFourHours);
+    setSettings(s);
 }
 
-void DateTimeSettings::handleTimeChanged(MeeGo::QmTime::WhatChanged what)
+bool DateTimeSettings::setSettings(Maemo::Timed::WallClock::Settings &s)
 {
-    switch (what) {
-    case MeeGo::QmTime::TimeChanged:
+    if (!s.check()) {
+        return false;
+    }
+
+    QDBusMessage reply = m_timed.wall_clock_settings_sync(s);
```

Comment:
How about using the async version? Nicer for qml, and 24h clock setting already was such.